components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/cache/AuthenticationContextCache.java

```diff
@@ -97,8 +111,41 @@ public void addToCache(AuthenticationContextCacheKey key, AuthenticationContextC
                             ", Operation : STORE ]";
                     log.debug("Authentication context is stored with details " + message);
                 }
+                if (isSessionDataStorageOptimizationEnabled && entry.getContext() != null) {
+                    try {
+                        AuthenticationContextLoader.getInstance().optimizeAuthenticationContext(entry.getContext());
+                    } catch (SessionDataStorageOptimizationClientException e) {
+                        if (log.isDebugEnabled()) {
+                            log.debug(String.format("Client error occurred while optimizing the Authentication " +
+                                    "context with context id: %s", entry.getContext().getContextIdentifier()), e);
+                        }
+                        return;
+                    } catch (SessionDataStorageOptimizationServerException e) {
+                        log.error("Server error occurred while optimizing the Authentication context with " +
+                                "context id: " + entry.getContext().getContextIdentifier(), e);
+                        return;
+                    } catch (SessionDataStorageOptimizationException e) {
+                        log.debug("Error occurred while optimizing the Authentication context with " +
+                                "context id: " + entry.getContext().getContextIdentifier(), e);
+                        return;
+                    }
+                }
                 SessionDataStore.getInstance().storeSessionData(key.getContextId(), AUTHENTICATION_CONTEXT_CACHE_NAME,
                         entry, tenantId);
+                try {
+                    AuthenticationContextLoader.getInstance().loadAuthenticationContext(entry.getContext());
+                } catch (SessionDataStorageOptimizationClientException e) {
+                    if (log.isDebugEnabled()) {
+                        log.debug(String.format("Client error occurred while loading optimized authentication context"
```

Comment:
Let's remove string format here. As it is only 1 param and also since its the last param

Reply:
Resolved with [`87d8c01`](https://github.com/wso2/carbon-identity-framework/pull/4446/commits/87d8c01eaa49ef513f71a55b77154408818a31b3)